### DISCUSSION TITLE
Sometimes the autocontain configuration is not there

### DISCUSSION THREAD

**Discussion by make-github-pseudonymous-again** [tridactyl/tridactyl#3655] [Q&A] [answered]
Sometimes all my tabs open in the default firefox container. When this happens,`:viewconfig` shows an empty `autocontain` (`"autocontain": {}`). Other than that, tridactyl appears to be working fine (shortcuts, theming, etc.).
I usually can workaround the problem by sourcing (`:source`) one or more times. Is anybody else experiencing this?

**Comment by bovine3dom** [comment, marked as answer]
Could be related to https://github.com/tridactyl/tridactyl/issues/1409

Do you use `:sanitise` at the top of your RC file? That seems to be the thing that causes #1409 - I'd recommend removing it.

**Reply by make-github-pseudonymous-again** [reply]
I do not use `:sanitise`. Here is the source if you are interested https://github.com/make-github-pseudonymous-again/dotfiles/blob/8b8c5b26c8f99d6e119f69c12a668eb88213ac88/.config/tridactyl/tridactylrc

**Reply by bovine3dom** [reply]
`:unset autocontain` is fairly similar to `:sanitise` - do you know why you have that line? I'd try removing it

**Reply by make-github-pseudonymous-again** [reply]
Maybe to disable some default `autocontain` setting? I do not remember. I could try removing it and see how it goes.

**Reply by bovine3dom** [reply]
It's empty by default: https://github.com/tridactyl/tridactyl/blob/12a07248d2c747a30a443c86322d895384fcb569/src/lib/config.ts#L527

Good luck : )

**Reply by make-github-pseudonymous-again** [reply]
The problem did not occur since I removed the line `:unset autocontain`.

Does removing this line mean that if I source a config that removes some of the autocontain declarations, these removed declarations will still have an effect until the next browser restart?

**Reply by bovine3dom** [reply]
> Does removing this line mean that if I source a config that removes some of the autocontain declarations, these removed declarations will still have an effect until the next browser restart?

Could you rephrase this? I don't understand your question :)

I think the answer is no.

**Reply by make-github-pseudonymous-again** [reply]
> Could you rephrase this? I don't understand your question :)

Is sourcing in tridactyl the same thing as sourcing in bash? Does it reset the config state when you source or does it patch the sourced file changes on top of the current config state? I have a `.tridactylrc` with a line that modifies the config state. What happens if I remove that line then source this file? Does the modification made by that line disappear as soon as I source the edited file?

**Reply by bovine3dom** [reply]
> Is sourcing in tridactyl the same thing as sourcing in bash?

Yes, it runs each command in the file one after the other

> Does it reset the config state when you source or does it patch the sourced file changes on top of the current config state? 

It patches it

> I have a .tridactylrc with a line that modifies the config state. What happens if I remove that line then source this file?

All the commands in that file will be executed apart from the line you removed.

> Does the modification made by that line disappear as soon as I source the edited file?

No. All settings in Tridactyl persist as soon as they are set. You can use `unset` or `sanitise` to clear them. Due to bugs it is a bad idea to use those in RC files. Run them manually when you need them.

**Reply by make-github-pseudonymous-again** [reply]
OK. Thanks! I understand the behavior now. I will avoid putting `unset` and `sanitise` commands in config files. I will only run them manually in the rare cases where it is needed.
